File: plugins/core/scripts/read_pdfs.py
import os
import sys
sys.stdout.reconfigure(encoding='utf-8')

from PyPDF2 import PdfReader

data_dir = os.path.join(os.path.dirname(__file__), "..", "01_Discovery", "data_collected")

def read_pdf(file_path):
    try:
        reader = PdfReader(file_path)
        text = ""
        for page in reader.pages:
            text += page.extract_text() + "\n"
        return text
    except Exception as e:
        return f"Error reading {file_path}: {e}"

def main():
    for file in os.listdir(data_dir):
        if file.endswith('.pdf'):
            print("\n" + "=" * 80)
            print(f"FILE: {file}")
            print("=" * 80 + "\n")
            content = read_pdf(os.path.join(data_dir, file))
            print(content)

if __name__ == "__main__":
    main()

File: plugins/core/scripts/read_pdfs.js
const fs = require("fs");
const path = require("path");
const pdfParse = require("pdf-parse/lib/pdf-parse");

const dataDir = path.join(__dirname, "..", "01_Discovery", "data_collected");

async function readPdf(filePath) {
    try {
        const dataBuffer = fs.readFileSync(filePath);
        const data = await pdfParse(dataBuffer);
        return data.text;
    } catch (error) {
        return `Error reading ${filePath}: ${error.message}`;
    }
}

async function readAllPdfs() {
    const files = fs.readdirSync(dataDir).filter(f => f.endsWith('.pdf'));

    for (const file of files) {
        console.log("\n" + "=".repeat(80));
        console.log(`FILE: ${file}`);
        console.log("=".repeat(80) + "\n");

        const content = await readPdf(path.join(dataDir, file));
        console.log(content);
    }
}

readAllPdfs().catch(console.error);
